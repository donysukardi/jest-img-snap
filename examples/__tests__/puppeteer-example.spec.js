// eslint is looking for `puppeteer` at root level package.json
// eslint-disable-next-line import/no-unresolved
const puppeteer = require('puppeteer');

describe('jest-img-snapshot with puppeteer', () => {
  let browser;

  beforeAll(async () => {
    browser = await puppeteer.launch();
  });

  it('works', async () => {
    const page = await browser.newPage();
    await page.goto('https://example.com');
    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot();
  });

  afterAll(async () => {
    await browser.close();
  });
});
