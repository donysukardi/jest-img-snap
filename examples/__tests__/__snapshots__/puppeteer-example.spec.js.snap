// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-image-snapshot usage with an image received from puppeteer works 1`] = `
Object {
  "checksum": "9c836c3c365355766335ee9786fc8125",
  "filename": "jest-image-snapshot-usage-with-an-image-received-from-puppeteer-works-1-snap.png",
  "type": "image",
}
`;
